src/models/iceflow/SIA2D/SIA2D_utils.jl

```diff
@@ -187,16 +223,16 @@ function SIA2D(
     S = B .+ H
 
     # Compute D in case is not provided in the simulation
-    if SIA2D_model.D_is_provided
-        D = SIA2D_model.D
+    if SIA2D_cache.D_is_provided
+        D = SIA2D_cache.D
```

Comment:
Looks good to me to have this flag here in `SIA2D_cache` rather than `SIA_model`, but keep in mind this will break the `Target` object defined in ODINN, so these changes should also being made there.